server/mscalendar/availability.go

```diff
@@ -155,14 +134,135 @@ func (m *mscalendar) retrieveUsersToSync(userIndex store.UserIndex, syncJobSumma
 	return users, calendarViews, nil
 }
 
+func (m *mscalendar) retrieveUsersToSyncUsingGoroutines(ctx context.Context, userIndex store.UserIndex, syncJobSummary *StatusSyncJobSummary, concurrency int) ([]*store.User, []*remote.ViewCalendarResponse, error) {
+	start := time.Now().UTC()
+	end := time.Now().UTC().Add(calendarViewTimeWindowSize)
+
+	numberOfLogs := 0
+	users := []*store.User{}
+	calendarViews := []*remote.ViewCalendarResponse{}
+	for _, u := range userIndex {
+		user, err := m.Store.LoadUser(u.MattermostUserID)
+		if err != nil {
+			syncJobSummary.NumberOfUsersFailedStatusChanged++
+			if numberOfLogs < logTruncateLimit {
+				m.Logger.Warnf("Not able to load user %s from user index. err=%v", u.MattermostUserID, err)
+			} else if numberOfLogs == logTruncateLimit {
+				m.Logger.Warnf(logTruncateMsg)
+			}
+			numberOfLogs++
+			continue
+		}
+
+		// If user does not have the proper features enabled, just go to the next one
+		if !(user.Settings.UpdateStatus || user.Settings.ReceiveReminders) {
+			continue
+		}
+
+		users = append(users, user)
+	}
```

Comment:
Not blocking: What I liked about the version was that the workers already start while more users are getting loaded. It's not a big deal, but why should the works wait until all users have been loaded?

Reply:
- I wanted to spawn the workers only if there was actually work to do (`len(users) > 0`)
- This loop would need to be in a goro as well, and it has some other data transfer that need to be taken care of (`users`, `numberOfLogs`, `syncJobSummary`). It would make this function more complex.